src/applications/personalization/rated-disabilities/components/TotalRatedDisabilities.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
+import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
+import PercetnageCalloutBox from '../components/PercentageCalloutBox';
+
+const TotalRatedDisabilities = props => {
+  const { loading, totalDisabilityRating, error } = props;
+  let content;
+  // If the data from the parent is loading ( loading prop ), show a loading indicator
+  // If there is an error, display an error message,
+  // If there is no rating, display a no rating message
+  // If there is a rating, display the rating and content
+  if (loading) {
+    content = (
+      <LoadingIndicator message="Loading your total disability rating..." />
+    );
+  } else if (error) {
+    const message = (
+      <span>
+        <p>
+          We're sorry. An error occurred when accessing your disability rating
+          information.
+        </p>
+        <h4>What you can do</h4>
+        <p>
+          Sign out of VA.gov, then log back in to try this page again. If the
+          error continues, please call the VA.gov Help Desk at 1-855-574-7286
+          (TTY:1-800-829-4833). We're here Monday-Friday, 8:00 a.m. - 8:00 p.m.
+          (ET).
+        </p>
+      </span>
+    );
+    content = (
+      <AlertBox
+        headline="Total disabilities error"
+        content={message}
+        status="error"
+        isVisible
+      />
+    );
+  } else if (!totalDisabilityRating) {
+    const message = (
+      <span>
+        <p>We don't have a disability rating for you in our system.</p>
+        <h4>Want to add disabilities?</h4>
+        <p>
+          If you believe you have disabilities related to your military service,
+          you can submit a claim to update your disability rating.
+        </p>
+        <a href="#" className="usa-link">
+          Submit a new claim
+        </a>
+      </span>
+    );
+    content = (
+      <AlertBox
+        headline="No disability rating to show"
+        content={message}
+        status="info"
+        isVisible
+      />
+    );
+  } else {
+    content = (
```

Comment:
**Suggestion:** Pull the contents of each of these control flow branches into a separate function so the body of `TotalRatedDisabilities` is easier to read.

```jsx
const errorMessage = () => ( ... );
const missingTotalMessage = () => ( ... );
const totalRatingMessage = () => ( ... );

const TotalRatedDisabilities = props => {
  let content;
  if (loading) {
    content = (
      <LoadingIndicator message="Loading your total disability rating..." />
    );
  } else if (error) {
    content = errorMessage();
  } else if (!totalDisabilityRating) {
    content = missingTotalMessage();
  } else {
    content = totalRatingMessage();
  }
  return <span>content</span>;
}
```

Reply:
Great feedback, addressed in https://github.com/department-of-veterans-affairs/vets-website/pull/10950/commits/605e2a3d0cfc25f9e941dbf7c81bc6337ea9ef67